### PR TITLE
CLI to support OW REST interfaces that do not require an auth header

### PR DESCRIPTION
**Pull request by mdeuser**
Update common HTTP request API to support REST interfaces that do not require an auth header

### CODE REVIEW

**Comment by mdeuser** [comment]
PG2 1421

**Comment by mdeuser** [comment]
@dubeejw - would you kindly review.  thanks!

File: tools/cli/go-whisk/whisk/action.go [hunk 1]
```diff
@@ -84,9 +84,9 @@ func (s *ActionService) List(packageName string, options *ActionListOptions) ([]
     }
     Debug(DbgError, "Action list route with options: %s\n", route)
 
-    req, err := s.client.NewRequestUrl("GET", routeUrl, nil, IncludeNamespaceInUrl, AppendOpenWhiskPathPrefix, EncodeBodyAsJson)
+    req, err := s.client.NewRequestUrl("GET", routeUrl, nil, IncludeNamespaceInUrl, AppendOpenWhiskPathPrefix, EncodeBodyAsJson, AuthRequired)
```
Comment (dubeejw) [inline tools/cli/go-whisk/whisk/action.go]: Looks like this code is repeated at least seven times throughout different files.

Comment (mdeuser) [inline tools/cli/go-whisk/whisk/action.go]: @dubeejw - Yeah, the line will look similar, but ahe actual `routeUrl` value will vary somewhat across these invocations.

Comment (dubeejw) [inline tools/cli/go-whisk/whisk/action.go]: Okay let's open an issue to deal with multiple occurrences of the same error messages.

Need a PG on this.

Comment (mdeuser) [inline tools/cli/go-whisk/whisk/action.go]: PG3 115 - only failures are the problematic kafka connector tests

**Comment by mdeuser** [comment]
@bjustin-ibm - ready for merging... @dubeejw performed actual review, but this PR needs a review record for which I do not have permissions to start that process...

**Pull request merged**